tests/utils/pv.go

```diff
@@ -78,6 +80,45 @@ func NewPVDefinition(pvName string, size string, labels map[string]string, stora
 	}
 }
 
+// NewBlockPVDefinition creates a PV definition with volueMode 'Block'
+func NewBlockPVDefinition(pvName string, size string, labels map[string]string) *k8sv1.PersistentVolume {
+	volumeMode := k8sv1.PersistentVolumeBlock
+	return &k8sv1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   pvName,
+			Labels: labels,
+		},
+		Spec: k8sv1.PersistentVolumeSpec{
+			AccessModes:                   []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce},
+			PersistentVolumeReclaimPolicy: k8sv1.PersistentVolumeReclaimDelete,
+			Capacity: k8sv1.ResourceList{
+				k8sv1.ResourceName(k8sv1.ResourceStorage): resource.MustParse(size),
+			},
+			VolumeMode: &volumeMode,
+			PersistentVolumeSource: k8sv1.PersistentVolumeSource{
+				Local: &k8sv1.LocalVolumeSource{
+					Path: "/mnt/local-storage/block-device/loop0",
+				},
+			},
+			NodeAffinity: &k8sv1.VolumeNodeAffinity{
+				Required: &k8sv1.NodeSelector{
+					NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+						{
+							MatchExpressions: []k8sv1.NodeSelectorRequirement{
+								{
+									Key:      "kubernetes.io/hostname",
+									Operator: k8sv1.NodeSelectorOpIn,
+									Values:   []string{"node01"},
```

Comment:
We can't assume that the container that created the block device is on node01, we have to look up the pod with FindPodByPrefix and determine which node it is running on (since that node will be the one with the block device)

Reply:
done